torch/testing/_internal/common_utils.py

```diff
@@ -807,10 +861,817 @@ def wrapper(*args, **kwargs):
+                ]):
+                    # TODO: this stuff should support storage
+                    # (well, maybe not batched)
+                    hit += 1
+                    return t.to("meta")
+                else:
+                    hit += 1
+                    r = meta_tensor(t)
+                    if type(t) is torch.nn.Parameter:
+                        r = torch.nn.Parameter(r, requires_grad=r.requires_grad)
+                    return r
+            elif torch.overrides.is_tensor_like(t):
+                # Blindly converting tensor subclasses to meta can cause
+                # unpredictable problems; e.g., FX tests will trace meta
+                # tensors into their trace / some subclasses don't correctly
+                # support meta.  Trying to YOLO this is more trouble than it's
+                # worth.
+                miss += 1
+                return t
+            else:
+                # non-Tensor types don't count as hit or miss
+                return t
+
+        do_meta = (
+            func not in skipped and
+            func not in meta_exclude_set and not torch.jit.is_tracing() and
+            not isinstance(func, torch.ScriptMethod)
+        )
+
+        if do_meta:
+            try:
+                meta_args = tree_map(to_meta, args)
+                meta_kwargs = tree_map(to_meta, kwargs)
+            except Exception as e:
+                raise RuntimeError(
+                    f"failed to convert args to meta; "
+                    f"originally (*{args}, **{kwargs})") from e
+
         r = func(*args, **kwargs)
+
+        # TODO: also handle cases where func raise an exception
+
+        # For now, only attempt if we managed to convert all tensor types
+        # (if any of them failed, we're in a mixed device situation and
+        # this isn't well supported)
+        if do_meta and hit > 0 and miss == 0:
+            try:
+                # suppress warnings
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    meta_r = func(*meta_args, **meta_kwargs)
+            except Exception as e:
+                suppress = False
+                if isinstance(e, NotImplementedError):
+                    m = RE_NOT_IMPLEMENTED_MSG.search(e.args[0])
+                    if m and m.group(1) not in ("aten::_efficientzerotensor", "aten::view_as_real"):
+                        skipped.add(func)
+                        print(f"\n    {torch.overrides.resolve_name(func) or func},  # MISSING {m.group(1)}")
+                        sys.stdout.flush()
+                        suppress = True
+                if not suppress:
+                    raise RuntimeError(f"""\
+failed to run: {torch.overrides.resolve_name(func) or func}(
+    *{meta_args},
+    **{meta_kwargs}
+  )""") from e
+            else:
+                pass
+                # self.test_case.assertEqual(meta_r.dtype, r.dtype)
```

Comment:
Not passing?

Reply:
Gonna turn these on later with fixes, but yes, autocast + meta is broken and fails this assert, because autocast rules are per device.